src/plugins/syscalls/syscalls.cpp

```diff
@@ -106,32 +106,63 @@
 #include <glib.h>
 #include <inttypes.h>
 #include "syscalls.h"
+#include "scproto.h"
 
 static event_response_t cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
+    int i,j;
 
-    syscalls *s = (syscalls*)info->trap->data;
+    syscall_wrapper_t *wrapper = (syscall_wrapper_t *)g_malloc(sizeof(syscall_wrapper_t));
+    wrapper = (syscall_wrapper*)info->trap->data;
+    syscalls *s = wrapper->sc;
 
-    switch(s->format) {
-    case OUTPUT_CSV:
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+
+    access_context_t ctx;
+    ctx.translate_mechanism = VMI_TM_PROCESS_DTB;
+    ctx.dtb = info->regs->cr3;
+    ctx.addr = info->regs->rsp+4;  // jump over base pointer
+
+    unsigned int nargs = syscall_struct[wrapper->syscall_index].num_args;
+
+    // multiply num args by 4 for 32 bit systems to get the number of bytes we need
+    // to read from the stack (only valid for 32 bit systems, 64 bit systems have
+    // some arguments on cx, dx, r8, r9, and the stack).  assumes standard calling
+    // convention (cdecl) for the visual studio compile.
+    unsigned int size = 4 * nargs; 
+    
+    unsigned char buf[size];
+
+    vmi_read(vmi, &ctx, buf, size);
+
+      switch(s->format) {
+      case OUTPUT_CSV:
         printf("syscall,%" PRIu32" 0x%" PRIx64 ",%s,%" PRIi64 ",%s,%s\n",
                info->vcpu, info->regs->cr3, info->procname, info->userid, info->trap->breakpoint.module, info->trap->name);
         break;
-    default:
-    case OUTPUT_DEFAULT:
-        printf("[SYSCALL] vCPU:%" PRIu32 " CR3:0x%" PRIx64 ",%s %s:%" PRIi64" %s!%s\n",
+      default:
+      case OUTPUT_DEFAULT:
+        printf("[SYSCALL] vCPU:%" PRIu32 " CR3:0x%" PRIx64 ",%s %s:%" PRIi64" %s!%s,Arguments:%d\n",
                info->vcpu, info->regs->cr3, info->procname,
                USERIDSTR(drakvuf), info->userid,
-               info->trap->breakpoint.module, info->trap->name);
+               info->trap->breakpoint.module, info->trap->name, nargs);
+        for(i=0;i<nargs;i++) {
+          printf("\t%s %s %s:0x",syscall_struct[wrapper->syscall_index].args[i].dir,syscall_struct[wrapper->syscall_index].args[i].type,syscall_struct[wrapper->syscall_index].args[i].name);
+          for(j=3;j>=0;--j) { // j must be signed
+            printf("%02X", buf[i*4+j]);
+          }
+          printf("\n");
+        }
         break;
-    }
-
+      }
+    drakvuf_release_vmi(drakvuf);
+//    g_free(wrapper); // segfaults. why?
```

Comment:
Because you overwrite the pointer previously. The wrapper shouldn't be freed here anyway, only in the plugin destructor.